### PR TITLE
8.2.2.1_update_teachers_add_check_email_kozhukh

### PR DESCRIPTION
**Pull request by OksanaKozhukh**

### CODE REVIEW

File: src/modules/teachers/teachers.service.ts [hunk 1]
```diff
@@ -126,8 +127,18 @@ class TeachersService {
         return teacher;
     }
 
-    public async updateOneOrThrow(id: number, data: ITeachersData, user: Teachers): Promise<Teachers> {
+    public async updateOneOrThrow(id: number, data: ITeachersData, user: Teachers):
+        Promise<Teachers> {
         return sequelize.transaction(async (transaction: Transaction) => {
+
+            if (await this.findOneByEmail(user.email)) {
```
Comment (Alan18081) [inline src/modules/teachers/teachers.service.ts:134]: You have to check not the user's email but the email from `data` object. And you should do that check only if the email is provided as a part of the update

Comment (OksanaKozhukh) [reply src/modules/teachers/teachers.service.ts]: done

File: src/modules/teachers/teachers.service.ts [hunk 2]
```diff
@@ -126,8 +127,18 @@ class TeachersService {
         return teacher;
     }
 
-    public async updateOneOrThrow(id: number, data: ITeachersData, user: Teachers): Promise<Teachers> {
+    public async updateOneOrThrow(id: number, data: ITeachersData, user: Teachers):
+        Promise<Teachers> {
         return sequelize.transaction(async (transaction: Transaction) => {
+
+            if (data.email && await this.findOneByEmail(data.email)) {
```
Comment (Alan18081) [inline src/modules/teachers/teachers.service.ts:134]: It is better to have one outer if statement `if(data.email) {....}`

Comment (OksanaKozhukh) [reply src/modules/teachers/teachers.service.ts]: done

File: src/modules/students/students.service.ts [hunk 3]
```diff
@@ -94,6 +94,14 @@ class StudentsService {
 
     public async updateOneOrThrow(id: number, data: Partial<IStudentsData>, user: Students): Promise<Students> {
         return sequelize.transaction(async (transaction) => {
+
+            if (data.email) {
+                if (await this.findOneByEmail(data.email) ||
+                    teachersService.findOneByEmail(data.email)) {
```
Comment (Alan18081) [inline src/modules/students/students.service.ts:100]: It should have `await` before `teachersService.findOneByEmail`

Comment (OksanaKozhukh) [reply src/modules/students/students.service.ts]: done

**Pull request merged**